### ISSUE TITLE
jbang devmode not honoring properties

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#44400]
### Describe the bug

I think since https://github.com/quarkusio/quarkus/pull/43160 `jbang -Dquarkus.dev some.java` does not honor Q:CONFIG.

```
///usr/bin/env jbang "$0" "$@" ; exit $?
//DEPS io.quarkus.platform:quarkus-bom:3.16.1@pom
//DEPS io.quarkus:quarkus-picocli

//Q:CONFIG quarkus.banner.enabled=false
//Q:CONFIG quarkus.log.level=ERROR
import io.quarkus.logging.Log;
import picocli.CommandLine.Command;

@Command
public class basicdev implements Runnable {
  


    @Override
    public void run() {
        System.out.println("You should see ERROR, but no WARN or INFO");
        Log.warn("This is a WARN message");
        Log.info("This is an INFO message");
        Log.error("This is an ERROR message");
    }
}
```

run above using `jbang --fresh basicdev.java` and it will only print ERROR logging, but run it with `jbang --fresh -Dquarkus.dev basicdev.java` it prints warn,info and error. As if it completely ignores Q:CONFIG

**Comment by radcortez** [comment]
Ok, it seems that Dev Mode users a different path:
https://github.com/quarkusio/quarkus/blob/e12a3776707f1c33dbecfd262b26d6b0f411a9a8/core/launcher/src/main/java/io/quarkus/launcher/JBangIntegration.java#L27-L48

**Comment by maxandersen** [comment]
well that for sure explains why its not honoring any config as it returns early.

Now I'm just wondering why it ever worked ...and how we can go about passing in Q:CONFIG properties in devmode...if at all.

**Comment by radcortez** [comment]
It may have worked before that code? 

I'm proposing a fix in [#44415](https://github.com/quarkusio/quarkus/pull/44415), which requires us to parse the source file again because we lose the comments section parameter.

Also, can you confirm that the hot reload is working? When I change the file, I get an error: `Cannot initialize file manager`.

**Comment by aloubyansky** [comment]
I guess we are lacking some integration tests
